packages/SwingSet/src/vats/vat-admin/vat-vat-admin.js

```diff
@@ -318,9 +318,7 @@ export function buildRootObject(vatPowers, _vatParameters, baggage) {
     noteRunningVat(vatID);
 
     const adminNode = makeAdminNode(vatID);
-    return E.when(pendingP, root => {
-      return { adminNode, root };
-    });
+    return E.when(pendingP, root => harden({ adminNode, root }));
```

Comment:
Note to self: this is a tidying change, and doesn't affect the behavior (`createVat()` continues to return a Promise that fires with a hardened `{ adminNode, root }` record, both of which are Presences, not promises). It's just that the new version makes the `harden()` explicit, and is more compact (and I didn't know the right `root => ({ adminNode, root })` syntax to let fat-arrow bodies return objects before). Promise resolution values get hardened by our liveslots layer before marshalling, else we'd get marshalling errors in a hard-to-report place, but it's always nice to be explicit about it.

Some of the other commits in this PR, like the one in `fakeVatAdmin.js`, *are* behavior changing, specifically they make the fake version behave like the real one, whereas the fake version currently returns a promise for a record whose `.root` property is a Promise, not a Presence, requiring an additional round of `E.when` or `.then` or `deeplyFulfilledObject` to resolve.

If this vat-vat-admin commit *were* behavior-changing, we'd need to have a conversation about deployment and backwards compatibility. The mainnet chain's kernel (or any other deployed swingset kernel) already has a vat-vat-admin, whose behavior will only change when explicitly upgraded. Other vats can't know exactly when this happens, so we'd need to ensure that they can deal with both the old and new behavior. We frequently have PRs which change both sides of an interface boundary at the same time, but of course that only works if both sides actually change, which really only happens for unit tests.

I'm not sure whether to recommend that this file's change be reverted, in the name of "don't add confusion between what's on mainnet and what's on trunk", or to recommend that we land it, in the name of "make the code incrementally better, so when we finally do make a real change and have a plan to deploy it, we won't forget to include this improvement". Probably the latter, but maybe we should put it in a separate commit, with a commit message to a future release manager, to let them know that this particular change doesn't cause an incompatibility, and does not need to provoke a vat-vat-admin upgrade, but instead can be rolled into whatever future upgrade we *do* put energy into deploying.

Reply:
Moved to separate PR at https://github.com/Agoric/agoric-sdk/pull/9558. Done.